application/src/main/java/org/togetherjava/tjbot/features/moderation/history/PurgeHistoryCommand.java

```diff
@@ -0,0 +1,175 @@
+import org.togetherjava.tjbot.features.CommandVisibility;
+import org.togetherjava.tjbot.features.SlashCommandAdapter;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static org.togetherjava.tjbot.db.generated.Tables.MESSAGE_HISTORY;
+
+public final class PurgeHistoryCommand extends SlashCommandAdapter {
+
+    private static final Logger logger = LoggerFactory.getLogger(PurgeHistoryCommand.class);
+    private static final String USER_OPTION = "user";
+    private static final String REASON_OPTION = "reason";
+    private static final String COMMAND_NAME = "purge-in-channel";
+    private static final int PURGE_MESSAGES_DEFAULT_DURATION =
+            Duration.PURGE_HISTORY_MIN_DURATION.getHours();
+    private static final String DURATION = "duration";
+    private final Database database;
+
+    /**
+     * Creates an instance of the command.
+     *
+     * @param database the database to retrieve message records of a target user
+     */
+
+    public PurgeHistoryCommand(Database database) {
+        super(COMMAND_NAME, "purge message history of user in same channel",
+                CommandVisibility.GUILD);
+
+        String optionUserDescription = "user who's message history you want to purge within %s hr"
+            .formatted(PURGE_MESSAGES_DEFAULT_DURATION);
+
+        String optionDurationDescription =
+                "duration in hours (default: %s)".formatted(PURGE_MESSAGES_DEFAULT_DURATION);
+
+        OptionData durationData =
+                new OptionData(OptionType.INTEGER, DURATION, optionDurationDescription, false);
+        durationData.addChoice("last 3 hrs", Duration.PURGE_HISTORY_THREE_HOURS.getHours())
+            .addChoice("last 6 hrs", Duration.PURGE_HISTORY_SIX_HOURS.getHours())
+            .addChoice("last 12 hrs", Duration.PURGE_HISTORY_TWELVE_HOURS.getHours())
+            .addChoice("last 24 hrs", Duration.PURGE_HISTORY_TWENTY_FOUR_HOURS.getHours());
+
+        getData().addOption(OptionType.USER, USER_OPTION, optionUserDescription, true)
+            .addOption(OptionType.STRING, REASON_OPTION,
+                    "reason for purging user's message history", true)
+            .addOptions(durationData);
+        this.database = database;
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        event.deferReply(true).queue();
+
+        OptionMapping targetOption =
+                Objects.requireNonNull(event.getOption(USER_OPTION), "target is null");
+        Member author = Objects.requireNonNull(event.getMember(), "author is null");
+        String reason = Objects.requireNonNull(event.getOption(REASON_OPTION).getAsString(),
+                "reason is null");
+
+        OptionMapping durationMapping = event.getOption(DURATION);
+        int duration = durationMapping == null ? PURGE_MESSAGES_DEFAULT_DURATION
```

Comment:
so when you trigger the command, default duration is 1 hr for ease of use so you don't have to specify duration. But if that value is provided, we prioritize that  duration instead . So it can be null at times and can also have an option.